webpack/src/components/Analogies.vue

```diff
@@ -7,15 +7,42 @@ import { retrieveAnalogiesLayer } from '../composables/storage';
 import Analogy from './Analogy.vue'
 
 export default {
-  components: {
-    Analogy
+  components: { Analogy },
+  props: {
+    dafaultDataType: {
+      type: String,
+      default: "co2"
+    },
+    hideTypeChange: {
+      type: Boolean,
+      default: false
+    },
+    analogy: {
+      type: String,
+      default: ""
+    }
   },
-  setup() {
+  setup(props) {
     const { t } = useI18n({});
-    let dataType = ref('co2');
+    const dataType = ref(props.dafaultDataType);
+    const hideTypeChange = ref(props.hideTypeChange);
+    const analogy = ref(props.analogy);
```

Comment:
more concise notation to create refs from props
```suggestion
    const { dataType , hideTypeChange , analogy } = toRefs(props);

```

Reply:
please also update for History, I missed that part in my previous review